### DISCUSSION TITLE
Can't set value of a child

### DISCUSSION THREAD

**Discussion by themachho** [zeux/pugixml#449] [Q&A] [answered]
Hello. Im trying to build a simple app. I have this xml template that i need to fill before send it to a web service

```
<?xml version="1.0" encoding="UTF-8"?>
<LoginRequest version="1.0">
    <header> 
        <Id></Id>
        <generationTime></generationTime>
        <expirationTime></expirationTime>
    </header>  
</LoginRequest>
```

This is my code:

```C++
pugi::xml_document doc;
pugi::xml_parse_result result = doc.load_file("xml_templates\\login_request.xml");
if (!result)
    return -1;

pugi::xml_node node = doc.child("LoginRequest").child("header").child("Id");
std::cout << node.value() << std::endl;
std::cout << node.set_value("12414412414");
std::cout <<  node.value() << std::endl;
```
My output is empty, 0, empty
```

0

```
So the xml is never modified. Im using vscode and in the debugger section, the values have <NULL> <Unable to read memory>.

The example modify_base works fine for me so i dont know what im doing wrong since the doc is loaded correcly and the set_value also works since 0 is the output. 
Even having the template with some dummy info i get the same output.

**Comment by themachho** [comment]
Ok, so i managed to do this:

pugi::xml_node node = doc.child("LoginRequest").child("header").child("Id").first_child();
std::cout << node.value() << std::endl;
std::cout << node.set_value("12414412414");
std::cout <<  node.value() << std::endl;

Now it works, but still dont understand why i have to acces the "first_child" of my node to modify the value. Another option is node.child_value() but to modify it i have to do first_child() first.

EDIT: so aparently this is because i have an empty tag <Id></Id> (access normal use) in the template vs when i have <Id>dummy_value</Id> (acces with first_child()). Any ideas on how to avoid this?

**Comment by zeux** [comment, marked as answer]
https://pugixml.org/docs/manual.html#dom.tree
https://pugixml.org/docs/manual.html#modify.text
